src/Microsoft.ML.ImageAnalytics/Imager.cs

```diff
@@ -0,0 +1,420 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using SkiaSharp;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// Provide interfaces for imaging operations.
+    /// </summary>
+    public class Imager : IDisposable
+    {
+        private SKBitmap _image;
+
+        /// <summary>
+        /// Create a new imager instance from a stream.
+        /// </summary>
+        /// <param name="imageStream">The stream to create the image from.</param>
```

Comment:
Make sure to call out the lifecyle of the stream.  Does this type own it and become responsible for disposing it?

Alternatively, does this just synchronously read from the stream during the constructor and it's done with it after that?

I couldn't tell from the implementation (nor the docs).

Reply:
The `SKBitmap` reads the stream/file synchronously and doesn't hold any reference to the stream/files after constructing the object. We shouldn't work about the lifecycle of the stream/file after that as we don't own it. 

> If so, would we also want an async factory API?

I am not sure why we'll need that. Note that what I am exposing here is only to satisfy the ML.NET scenarios. If we find any need in the future to have async calls for it, we can do that. 

> I think if the constructor is doing any substantial work, then we should instead create factory methods and move that work to those methods. On the other hand, if it's just initializing and then the work happens later we should document that.

What is the difference if we have it as constructor or factory method? The constructor I am exposing is just a wrapper for instantiating the `SKBitmap` object which fully initializes the object (meta properties and pixels buffer). I don't mind changing the constructors to factory methods, just want to understand what the difference would be.

Reply:
done.